core/src/main/java/org/apache/iceberg/stats/StatsFormat.java

```diff
@@ -0,0 +1,162 @@
+import io.airlift.compress.zstd.ZstdDecompressor;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableTable;
+import org.apache.iceberg.relocated.com.google.common.collect.Table;
+import org.apache.iceberg.util.ByteBuffers;
+
+public final class StatsFormat {
+  private StatsFormat() {
+  }
+
+  enum Flag {
+    FOOTER_PAYLOAD_COMPRESSED(0, 0),
+    /**/;
+
+    private static final Table<Integer, Integer, Flag> byByteAndBit = Stream.of(values())
+        .collect(ImmutableTable.toImmutableTable(Flag::byteNumber, Flag::bitNumber, Function.identity()));
+
+    private final int byteNumber;
+    private final int bitNumber;
+
+    Flag(int byteNumber, int bitNumber) {
+      this.byteNumber = byteNumber;
+      this.bitNumber = bitNumber;
+    }
+
+    @Nullable
+    static Flag fromBit(int byteNumber, int bitNumber) {
+      return byByteAndBit.get(byteNumber, bitNumber);
+    }
+
+    public int byteNumber() {
+      return byteNumber;
+    }
+
+    public int bitNumber() {
+      return bitNumber;
+    }
+  }
+
+  static final int FOOTER_MAGIC_OFFSET = 0;
+  static final int FOOTER_STRUCT_LENGTH = 12;
+  static final int FOOTER_STRUCT_PAYLOAD_SIZE_OFFSET = 0;
+  static final int FOOTER_STRUCT_FLAGS_OFFSET = 4;
+  static final int FOOTER_STRUCT_FLAGS_LENGTH = 4;
+  static final int FOOTER_STRUCT_MAGIC_OFFSET = 8;
+  static final StatsCompressionCodec FOOTER_COMPRESSION_CODEC = StatsCompressionCodec.LZ4;
+
+  public static final String CREATED_BY_PROPERTY = "created-by";
+
+  static byte[] getMagic() {
+    return new byte[] {0x50, 0x46, 0x42, 0x31};
+  }
+
+  static void writeIntegerLittleEndian(OutputStream outputStream, int value) throws IOException {
+    outputStream.write(0xFF & value);
+    outputStream.write(0xFF & value >> 8);
+    outputStream.write(0xFF & value >> 16);
+    outputStream.write(0xFF & value >> 24);
+  }
+
+  static int readIntegerLittleEndian(byte[] data, int offset) {
+    return Byte.toUnsignedInt(data[offset]) |
+        (Byte.toUnsignedInt(data[offset + 1]) << 8) |
+        (Byte.toUnsignedInt(data[offset + 2]) << 16) |
+        (Byte.toUnsignedInt(data[offset + 3]) << 24);
+  }
+
+  static ByteBuffer compress(StatsCompressionCodec codec, ByteBuffer input) {
+    if (codec == StatsCompressionCodec.NONE) {
+      return input.duplicate();
+    }
+    Compressor compressor = createCompressor(codec);
+    ByteBuffer output = ByteBuffer.allocate(compressor.maxCompressedLength(input.remaining()));
+    compressor.compress(input.duplicate(), output);
+    output.flip();
+    return output;
+  }
+
+  private static Compressor createCompressor(StatsCompressionCodec codec) {
+    switch (codec) {
+      case NONE:
+        break;
```

Comment:
Won't think fall through to the `throw UnsupportedOperationException`?

Reply:
i don't get it

Reply:
I guess this should never be called with `NONE`, but I'd expect this to return something like a `NoOpCompressor` instead of throwing an UnsupportedException.

Reply:
Yes, it's never called with `NONE`, so `NoOpCompressor` would be a dead code.
to make the code clearer, i refactored this a bit